<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ImageViewerClass</class>
 <widget class="QMainWindow" name="ImageViewerClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1015</width>
    <height>691</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ImageViewer</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QToolBox" name="toolBox">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>275</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="page2D">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>275</width>
         <height>558</height>
        </rect>
       </property>
       <attribute name="label">
        <string>2D</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QGroupBox" name="groupBox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>100</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>Color</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_2">
             <item>
              <widget class="QPushButton" name="pushButtonSetColor">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonClear">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Clear</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBoxDraw">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Draw</string>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="2" column="0" colspan="3">
            <widget class="QComboBox" name="comboBoxCurvedType">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <item>
              <property name="text">
               <string>Hermit</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Bezier</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Coon</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="1" column="0" colspan="3">
            <widget class="QComboBox" name="comboBoxLineAlg">
             <item>
              <property name="text">
               <string>DDA</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Bresenham</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QToolButton" name="toolButtonDrawCurved">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Curved</string>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QToolButton" name="toolButtonDrawCircle">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Circle</string>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QToolButton" name="toolButtonDrawLine">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Line</string>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
             <property name="autoRepeatDelay">
              <number>0</number>
             </property>
            </widget>
           </item>
           <item row="3" column="2">
            <widget class="QToolButton" name="toolButtonDrawPolygon">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Polygon</string>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBoxEdit">
          <property name="title">
           <string>Edit</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="0" column="0" rowspan="2">
            <layout class="QGridLayout" name="gridLayout_3">
             <item row="3" column="0">
              <widget class="QPushButton" name="pushButtonShear">
               <property name="text">
                <string>Shear</string>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QLabel" name="label_4">
               <property name="layoutDirection">
                <enum>Qt::LeftToRight</enum>
               </property>
               <property name="text">
                <string>Fact.</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
              </widget>
             </item>
             <item row="2" column="2">
              <widget class="QDoubleSpinBox" name="doubleSpinBoxScaleX">
               <property name="minimum">
                <double>0.500000000000000</double>
               </property>
               <property name="maximum">
                <double>2.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.250000000000000</double>
               </property>
              </widget>
             </item>
             <item row="4" column="0">
              <widget class="QPushButton" name="pushButtonMirror">
               <property name="text">
                <string>Mirror</string>
               </property>
              </widget>
             </item>
             <item row="3" column="2">
              <widget class="QDoubleSpinBox" name="doubleSpinBoxShear">
               <property name="minimum">
                <double>-5.000000000000000</double>
               </property>
               <property name="maximum">
                <double>5.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.250000000000000</double>
               </property>
               <property name="value">
                <double>1.250000000000000</double>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QLabel" name="label_2">
               <property name="layoutDirection">
                <enum>Qt::LeftToRight</enum>
               </property>
               <property name="text">
                <string>x,y</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
              </widget>
             </item>
             <item row="0" column="2">
              <widget class="QSpinBox" name="spinBoxRotate">
               <property name="buttonSymbols">
                <enum>QAbstractSpinBox::UpDownArrows</enum>
               </property>
               <property name="minimum">
                <number>-360</number>
               </property>
               <property name="maximum">
                <number>360</number>
               </property>
               <property name="singleStep">
                <number>45</number>
               </property>
               <property name="stepType">
                <enum>QAbstractSpinBox::DefaultStepType</enum>
               </property>
               <property name="value">
                <number>45</number>
               </property>
              </widget>
             </item>
             <item row="2" column="3">
              <widget class="QDoubleSpinBox" name="doubleSpinBoxScaleY">
               <property name="minimum">
                <double>0.500000000000000</double>
               </property>
               <property name="maximum">
                <double>2.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.250000000000000</double>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QPushButton" name="pushButtonScale">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Ignored">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Scale</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLabel" name="label">
               <property name="layoutDirection">
                <enum>Qt::LeftToRight</enum>
               </property>
               <property name="text">
                <string>Deg.</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QPushButton" name="pushButtonRotate">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="maximumSize">
                <size>
                 <width>100</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>Rotate</string>
               </property>
               <property name="default">
                <bool>false</bool>
               </property>
               <property name="flat">
                <bool>false</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBoxFill">
          <property name="title">
           <string>Fill</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="0" column="0">
            <layout class="QGridLayout" name="gridLayout_2">
             <item row="1" column="0">
              <widget class="QPushButton" name="pushButtonSetColorA"/>
             </item>
             <item row="1" column="1">
              <widget class="QPushButton" name="pushButtonSetColorB"/>
             </item>
             <item row="1" column="2">
              <widget class="QPushButton" name="pushButtonSetColorC"/>
             </item>
             <item row="0" column="0" colspan="3">
              <widget class="QComboBox" name="comboBoxFillType">
               <item>
                <property name="text">
                 <string>Solid color</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Nearest-neighbor</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Barycentric</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="comboBoxShowAddons">
          <item>
           <property name="text">
            <string>Lines + points</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Lines</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Points</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>None</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page3D">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>275</width>
         <height>558</height>
        </rect>
       </property>
       <attribute name="label">
        <string>3D</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QGroupBox" name="groupBoxSimples">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="title">
             <string>simples</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout">
             <item>
              <layout class="QGridLayout" name="gridLayout_7">
               <item row="0" column="2">
                <widget class="QPushButton" name="pushButtonUVSphere">
                 <property name="text">
                  <string>UV Sphere</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="3">
                <widget class="QPushButton" name="pushButtonCube">
                 <property name="text">
                  <string>Cube</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="2">
                <widget class="QSpinBox" name="spinBoxRings">
                 <property name="prefix">
                  <string>Rings: </string>
                 </property>
                 <property name="minimum">
                  <number>1</number>
                 </property>
                </widget>
               </item>
               <item row="1" column="2">
                <widget class="QSpinBox" name="spinBoxRadius">
                 <property name="prefix">
                  <string>Radius: </string>
                 </property>
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>500</number>
                 </property>
                 <property name="value">
                  <number>150</number>
                 </property>
                </widget>
               </item>
               <item row="1" column="3">
                <widget class="QSpinBox" name="spinBoxCubeSideLength">
                 <property name="suffix">
                  <string/>
                 </property>
                 <property name="prefix">
                  <string>Length: </string>
                 </property>
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>500</number>
                 </property>
                 <property name="value">
                  <number>150</number>
                 </property>
                </widget>
               </item>
               <item row="3" column="2">
                <widget class="QSpinBox" name="spinBoxSegments">
                 <property name="prefix">
                  <string>Segments: </string>
                 </property>
                 <property name="minimum">
                  <number>3</number>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBoxTools3D">
          <property name="layoutDirection">
           <enum>Qt::LeftToRight</enum>
          </property>
          <property name="title">
           <string>Tools</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_6">
           <item row="1" column="1">
            <widget class="QPushButton" name="pushButtonLoad">
             <property name="text">
              <string>Load object</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QSlider" name="horizontalSliderAzimuth">
             <property name="maximum">
              <number>360</number>
             </property>
             <property name="sliderPosition">
              <number>0</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="tickPosition">
              <enum>QSlider::NoTicks</enum>
             </property>
             <property name="tickInterval">
              <number>0</number>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QComboBox" name="comboBoxSaveRepresentation">
             <item>
              <property name="text">
               <string>Surface</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Wireframe</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Points</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="6" column="1">
            <widget class="QSlider" name="verticalSliderZenith">
             <property name="layoutDirection">
              <enum>Qt::LeftToRight</enum>
             </property>
             <property name="minimum">
              <number>-90</number>
             </property>
             <property name="maximum">
              <number>90</number>
             </property>
             <property name="sliderPosition">
              <number>0</number>
             </property>
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="tickInterval">
              <number>1</number>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QPushButton" name="pushButtonSave">
             <property name="text">
              <string>Save object</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QComboBox" name="comboBoxProjectType">
             <item>
              <property name="text">
               <string>Parallel</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Perspective</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QSpinBox" name="spinBoxDistance">
             <property name="prefix">
              <string>Distance: </string>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>500</number>
             </property>
             <property name="value">
              <number>50</number>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QScrollArea" name="scrollArea">
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>714</width>
         <height>616</height>
        </rect>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1015</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave_as"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuImage">
    <property name="title">
     <string>Image</string>
    </property>
    <addaction name="actionClear"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuImage"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave_as">
   <property name="text">
    <string>Save as</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionClear">
   <property name="text">
    <string>Clear</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Alt+F4</string>
   </property>
  </action>
  <action name="actionResize">
   <property name="text">
    <string>Resize</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="ImageViewer.qrc"/>
 </resources>
 <connections/>
</ui>
